### PR TITLE
Add `advanced_machine_features` to GCE Instance Templates

### PR DESCRIPTION
**Pull request by upodroid**
Part of https://github.com/hashicorp/terraform-provider-google/issues/9251 

Raised instance template separately as it is easier to work with. There is a bug in the instances update method.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `advanced_machine_features` fields to `google_compute_instance_template`
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
getting a build failure in tpc build process

```
# github.com/GoogleCloudPlatform/terraform-google-conversion/google [github.com/GoogleCloudPlatform/terraform-google-conversion/google.test]
google/compute_instance_helpers.go:356:3: unknown field 'ThreadsPerCore' in struct literal of type "google.golang.org/api/compute/v0.beta".AdvancedMachineFeatures
google/compute_instance_helpers.go:367:58: AdvancedMachineFeatures.ThreadsPerCore undefined (type *"google.golang.org/api/compute/v0.beta".AdvancedMachineFeatures has no field or method ThreadsPerCore)
FAIL	github.com/GoogleCloudPlatform/terraform-google-conversion/google [build failed]
FAIL

```

**Comment by upodroid** [comment]
Maybe enable dependabot on that repo?

**Review by ScottSuarez — APPROVED** [review]

File: mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb [hunk 1]
```diff
@@ -556,11 +556,38 @@ func resourceComputeInstanceTemplate() *schema.Resource {
 						"enable_confidential_compute": {
 							Type:        schema.TypeBool,
 							Required:    true,
+							ForceNew:    true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:559]: @ScottSuarez @upodroid: Why was this made `ForceNew`? That's appears to be a breaking change at first glance, and I can't see any context in this PR on why it wouldn't be in this circumstance.

Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:559]: It was something I forgot when I first added CVM to this resource. The resource doesn't have have an update method so it is safe to sneak this change through.

Comment (rileykarson) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:559]: Ack, thanks. If you're making similar changes in the future can you make sure to specifically call that out when you send the review? Showing that modifying the field always returns an error is quick to collect and does all we need to justify the change.

One case that gives me pause as a reviewer is when Terraform *doesn't* return an error. We may want to avoid the change in those cases, even if it's correct. Unfortunately not all users run `terraform plan` before apply, particularly after provider updates, and depending on the circumstances we may want to not make the change until a major version.

Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:559]: I'll bear that in mind. There isn't a fix for end users running apply carelessly and not pinning provider versions.

File: mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb [hunk 2]
```diff
@@ -556,11 +556,38 @@ func resourceComputeInstanceTemplate() *schema.Resource {
...
 							Description: `Defines whether the instance should have confidential compute enabled.`,
 						},
 					},
 				},
 			},
+			"advanced_machine_features": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:569]: @upodroid: Just curious, why did you choose to make the value `Computed: true`? I haven't seen the API return a default value, does it have a default that changes based on other configuration between the resource? (sorry for the drive-by comments here, there's a conversation internally about the implementation of the field)

Comment (karlkfi) [inline mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:569]: Does this need to be Computed=true?

It seems like this would cause undesirable behavior.

Going from
```
advanced_machine_features {
  threads_per_core = 1
  enable_nested_virtualization = true
}
```
to
```
// removed
```
would NOT trigger an update/recreate. It would just be ignored, even tho the `threads_per_core` default value is 0.

Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:569]: Values were computed in `google_compute_instance` so I copied them over. I did the bulk of the testing with `google_compute_instance`

Comment (rileykarson) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:569]: Thanks! And GCE Instance will return the value when not sent in some circumstances?

Comment (rileykarson) [reply mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:569]: Also see https://github.com/hashicorp/terraform-provider-google/issues/9436 if you're interested in some of the context here, I'm seeing @karlkfi decided to comment around the same time I did 🙂
